Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototypeFunctions.h

```diff
@@ -68,31 +68,69 @@
 
 namespace JSC {
 
+namespace JSGenericTypedArrayViewPrototypeFunctionsInternal {
+static constexpr bool verbose = false;
+}
+
+template<typename ViewClass>
+ALWAYS_INLINE bool speciesWatchpointIsValid(JSGlobalObject* globalObject, ViewClass* thisObject)
+{
+    auto* prototype = globalObject->typedArrayPrototype(ViewClass::TypedArrayStorageType);
+
+    if (globalObject->typedArraySpeciesWatchpointSet(ViewClass::TypedArrayStorageType).state() == ClearWatchpoint) {
+        globalObject->tryInstallTypedArraySpeciesWatchpoint(ViewClass::TypedArrayStorageType);
```

Comment:
If we only do this once, how do we guarantee that the `__proto__` of something like Uint8Array is TypedArray?

Reply:
`tryInstallTypedArraySpeciesWatchpoint` is checking this exact thing.

Reply:
And `PropertyCondition::absence` keeps guaranteeing this thing (see commit message for more detail) :)